dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04205MigrateVanityURLToContent.java

```diff
@@ -0,0 +1,249 @@
+				dc.addParam(title); // friendly_name
+				dc.addParam("8e850645-bb92-4fda-a765-e67063a59be0"); // structure_inode
+				dc.addParam(uuid); // identifier
+				dc.addParam(lang); // language_id
+				dc.addParam(title); // text1(title)
+				dc.addParam(site); // text2(site)
+				dc.addParam(uri); // text3(uri)
+				dc.addParam(oldUri); // text4(forwardTo)
+				dc.addParam(301); // integer1(action)
+				dc.addParam(0); // integer2(sort)
+				for (int i = 0; i < 48; i++) {
+					dc.addParam(0); // all other integer and float fields
+				}
+				for (int i = 0; i < 25; i++) {
+					dc.addParam(false); // all bool fields
+				}
+
+				dc.loadResult();
+
+				// Insert on Contentlet_Version_Info
+				dc.setSQL(INSERT_CONTENTLET_VERSION_INFO_QUERY);
+				dc.addParam(uuid); // identifier
+				dc.addParam(lang); // lang
+				dc.addParam(inode); // live_inode
+				dc.addParam(inode); // working_inode
+				dc.addParam(false); // deleted
+				dc.addParam(new Date()); // locked_on
+				dc.addParam(new Date()); // version_ts
+				dc.loadResult();
+			}
+		}
+
+	}
+	
+	/**
+	 * This method splits the old Url into the site and uri. 
+	 * Also make the call for the {@link #findHostId(String)} that returns the host identifier
+	 * 
+	 * @param url. The value of the url column on the virtual_link table.
+	 * @return a map with the host identifier and uri
+	 * @throws DotDataException
+	 */
+	private Map<String,String> splitURL(String url) throws DotDataException{
+		String hostId = Host.SYSTEM_HOST;
+		
+		int hostSplit = url.indexOf(":");
+		if(hostSplit != -1){
+			String hostName = url.substring(0, url.lastIndexOf(":"));
+			url = url.substring(hostSplit+2);
+			hostId = findHostId(hostName);
+		}
+		
+		return map("siteId", hostId, "uri", url);
+	}
+
+	/**
+	 * This method searches the site identifier, if it doesn't exists it returns null.
+	 * 
+	 * @param hostName
+	 * @return the identifier of the site
+	 * @throws DotDataException
+	 */
+	private String findHostId(String hostName) throws DotDataException {
+		String hostId = null;
+		DotConnect dc = new DotConnect();
+		dc.setSQL(SELECT_HOSTID_BY_NAME_QUERY);
+		dc.addParam(hostName);
+		List<Map<String,String>> result = dc.loadResults();
+		
+		if(!result.isEmpty()){
+			hostId = result.get(0).get("identifier").toString();
+		}
+		
+		return hostId;
+	}
+	
+	/**
+	 * This method searches the lang of the Site that the Vanity URL is referenced.
+	 * 
+	 * @param hostId
+	 * @return the lang of the site
+	 * @throws DotDataException
+	 */
+	private String findHostLang(String hostId) throws DotDataException{
+		DotConnect dc = new DotConnect();
+		dc.setSQL(SELECT_HOSTLANG_BY_ID_QUERY);
+		dc.addParam(hostId);
+		List<Map<String,String>> hostLang = dc.loadResults();
+		
+		return hostLang.get(0).get("language_id").toString();
+	}
+
+	@Override
+	public boolean forceRun() {
+		return true;
+	}
+
+	@Override
+	public String getPostgresScript() {
+		// TODO Auto-generated method stub
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)